### PR TITLE
Plugin generic evm ocr2

### PR DESCRIPTION
**Pull request by connorwstein**
Currently the transmitter and config tracker are specific to the median plugin when they need to be generic to support other plugin types. This removes the plugin specific geth wrapper code in favor of using the client interacting directly with OCR2Abstract functions https://github.com/smartcontractkit/libocr/blob/master/contract2/OCR2Abstract.sol#L7. In addition the config tracker is also used for the latest round requested feature which is also median specific, so I've pulled it out into its own component.

### CODE REVIEW

File: core/services/relay/evm/request_round_tracker.go [hunk 1]
```diff
@@ -0,0 +1,210 @@
...
+	offchain_aggregator_wrapper "github.com/smartcontractkit/chainlink/core/internal/gethwrappers2/generated/offchainaggregator"
+	"github.com/smartcontractkit/chainlink/core/logger"
+	"github.com/smartcontractkit/chainlink/core/services/ocrcommon"
+	"github.com/smartcontractkit/chainlink/core/services/pg"
+	"github.com/smartcontractkit/chainlink/core/utils"
+)
+
+// RequestRoundTracker complies with ContractConfigTracker interface and
+// handles log events related to the contract more generally
+type RequestRoundTracker struct {
```
Comment (connorwstein) [inline core/services/relay/evm/request_round_tracker.go:26]: this is identical to before except it doesn't also process the config set event

File: core/internal/features_ocr2_test.go [hunk 2]
```diff
@@ -14,6 +15,9 @@ import (
 	"testing"
 	"time"
 
+	"github.com/ethereum/go-ethereum/accounts/abi"
+	"github.com/smartcontractkit/chainlink/core/services/relay/evm"
```
Comment (jmank88) [inline core/internal/features_ocr2_test.go:19]: https://www.notion.so/chainlink/IDE-Settings-356dfa857ed54af99764b05a7a59082d#39df10b17ed341768702c2cbdd33026c

File: core/services/relay/evm/config_tracker.go [hunk 3]
```diff
@@ -0,0 +1,232 @@
...
+		FromBlock: new(big.Int).SetUint64(changedInBlock),
+		ToBlock:   new(big.Int).SetUint64(changedInBlock),
+	}
+	logs, err := c.client.FilterLogs(ctx, query)
+	if err != nil {
+		return ocrtypes.ContractConfig{}, err
+	}
+	if len(logs) == 0 {
+		c.lggr.Warnw("Contract not configured yet")
+		return ocrtypes.ContractConfig{}, err
```
Comment (jmank88) [inline core/services/relay/evm/config_tracker.go:130]: Should this be returning an error too? e.g.
```suggestion
		err := errors.New("contract not configured yet")
		c.lggr.Warnw(err)
		return ocrtypes.ContractConfig{}, err
```

File: core/services/relay/evm/request_round_tracker.go [hunk 4]
```diff
@@ -0,0 +1,209 @@
...
+		odb,
+		pg.NewQ(db, logger, chain),
+		ocrcommon.NewBlockTranslator(chain, ethClient, logger),
+		ctx,
+		cancel,
+		nil,
+		ocr2aggregator.OCR2AggregatorRoundRequested{},
+		sync.RWMutex{},
+		-1,
+		sync.RWMutex{},
```
Comment (jmank88) [inline core/services/relay/evm/request_round_tracker.go:82]: This is pretty large, maybe worth using named fields and dropping the zero values?

**Review by jmank88 — APPROVED** [review]

**Pull request merged**